pkg/kubectl/rolling_updater.go

```diff
@@ -411,13 +417,17 @@ func (r *RollingUpdater) readyPods(oldRc, newRc *api.ReplicationController) (int
 			return 0, 0, err
 		}
 		for _, pod := range pods.Items {
-			if api.IsPodReady(&pod) {
-				switch controller.Name {
-				case oldRc.Name:
-					oldReady++
-				case newRc.Name:
-					newReady++
-				}
+			if !api.IsPodReady(&pod) {
+				continue
+			}
+			if r.nowFn != nil && minReadySeconds > 0 && !deployment.IsPodAvailable(&pod, minReadySeconds, r.nowFn().Time) {
```

Comment:
I don't think you need the intermediate check (minReadySeconds > 0). It should be handled by IsPodAvailable, no? Same thing goes for the IsPodReady check above. It's already in IsPodAvailable.